tools/db/mls-users/src/MlsUsers/Types.hs

```diff
@@ -0,0 +1,123 @@
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE TemplateHaskell #-}
+
+-- This file is part of the Wire Server implementation.
+--
+-- Copyright (C) 2024 Wire Swiss GmbH <opensource@wire.com>
+--
+-- This program is free software: you can redistribute it and/or modify it under
+-- the terms of the GNU Affero General Public License as published by the Free
+-- Software Foundation, either version 3 of the License, or (at your option) any
+-- later version.
+--
+-- This program is distributed in the hope that it will be useful, but WITHOUT
+-- ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+-- FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+-- details.
+--
+-- You should have received a copy of the GNU Affero General Public License along
+-- with this program. If not, see <https://www.gnu.org/licenses/>.
+
+module MlsUsers.Types where
+
+import qualified Cassandra as C
+import Control.Lens
+import qualified Data.Aeson as A
+import qualified Data.Aeson.Encode.Pretty as A
+import qualified Data.ByteString.Lazy.Char8 as LC8
+import Data.Id
+import Data.Text.Strict.Lens
+import Database.CQL.Protocol (Record (..), TupleType, recordInstance)
+import Imports
+import Options.Applicative
+import Wire.API.User
+
+data UserRow = UserRow
+  { userId :: UserId,
+    activated :: Bool,
+    status :: Maybe AccountStatus,
+    supportedProtocols :: Set BaseProtocolTag
+  }
+  deriving (Generic)
+
+instance A.ToJSON UserRow
```

Comment:
Can you use deriving strategies here please?

Reply:
What do you mean? This is using generic deriving, it's how it's done in other places too.